### ISSUE TITLE
[vLLM] bad_words silently ignored in vllm_tt Sampler

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#3363]
## Issue

`bad_words` in `vllm.SamplingParams` has no effect on TT hardware. Words specified in `bad_words` appear freely in generated output. Tracked in the test suite as:

```python
@pytest.mark.xfail(reason="bad_words not enforced in vllm_tt Sampler")
def test_bad_words(llm, prompt): ...
```

## Root Cause

`XLASupportedSamplingMetadata` (forked from `vllm/v1/sample/tpu/metadata.py`) has `bad_words_token_ids = None` as an untyped class-level stub — it is never populated in `from_input_batch`. `vllm_tt/sampler.py` has no `apply_bad_words` method.

The upstream GPU implementation (`vllm/v1/sample/ops/bad_words.py`) applies bad words by iterating over the batch with Python loops, checking whether the tail of generated tokens matches a bad word prefix, then setting the next-token logit to `-inf`. This involves per-batch Python iteration and runtime token history access — neither compiles to a static XLA graph.

**Complication**: `bad_words` supports multi-token sequences. Suppressing `["New York"]` requires checking whether the last token was `"New"` before banning `"York"` — this prefix matching needs generated token history per request, which is engine-level state. Multi-token bad words are fundamentally harder to vectorize. Single-token bad words are tractable: they reduce to a dense `[batch, vocab]` bool mask set to `-inf`, the same pattern used by `apply_penalties` (#3331).

## Repro

```python
params = vllm.SamplingParams(temperature=0.0, max_tokens=16, bad_words=["the"])
output = llm.generate(prompt, params)[0].outputs[0].text
assert "the" not in output.lower()  # FAILS — "the" appears freely
```

## Suggested Fix

**Phase 1 — single-token bad words:**
- In `from_input_batch`, extract single-token entries from `bad_words_token_ids` and build a dense `[batch, vocab]` float32 additive mask (0 or `-inf`).
- Add `no_bad_words: bool = True` gate (same pattern as `no_penalties`) so the path is absent from compiled graphs when unused.
- Add `apply_bad_words` to `vllm_tt/sampler.py`: `logits += bad_words_mask`.

**Phase 2 — multi-token bad words:**
Requires prefix matching against generated token history — engine-level, not straightforward to vectorize for XLA. Defer or apply on CPU post-sampling.

**Comment by kmabeeTT** [comment]
Asserts were added via https://github.com/tenstorrent/tt-xla/pull/3367 so this no longer is silently ignored, but fails loudly.

**Comment by kmabeeTT** [comment]
"Phase 1 — single-token bad words" support merged just now via https://github.com/tenstorrent/tt-xla/pull/3415
